### PR TITLE
Mailchimp api v3

### PR DESCRIPTION
**Pull request by rclachance**

### CODE REVIEW

**Comment by benlk** [comment]
note that, because of the submodule, we'll have to do the first deploy of this by hand, because of https://github.com/INN/deploy-tools/issues/52

File: includes/actions.php [hunk 1]
```diff
@@ -1,254 +1,172 @@
...
+			'settings' => $campaign_content,
 			$segment_opts,
 			null
-		);
+		]);
 
-		if ( isset( $response['status'] ) && $response['status'] == 'error' ) {
-			rns_return_post_to_draft( $post_ID );
-			wp_die( 'Error: ' . $response['error'], 'Error', 'back_link=true' );
+		if ( isset( $response['status'] ) && 'error' === $response['status'] ) {
```
Comment (benlk) [inline includes/actions.php:154]: Why do the sides of the `===` operator switch here? I'm curious.

File: includes/actions.php [hunk 2]
```diff
@@ -1,254 +1,172 @@
...
+				rns_return_post_to_draft( $post_id );
+				wp_die( 'Error: ' . wp_kses( $sent['error'] ), 'Error', 'back_link=true' );
+			} else {
+				add_post_meta( $post_id, '_rns_transmission_sent', 1 );
 			}
 		}
-	}
-}
+	} // End if().
+} // End if().
```
Comment (benlk) [inline includes/actions.php]: This is the end of the function, not of one of the if()s

File: includes/admin.php [hunk 3]
```diff
@@ -218,31 +220,31 @@ function rns_transmissions_api_key_input() {
...
+	echo '<input type="text" name="rns_transmissions_options[from_name]" id="from_name" value="' . esc_attr( $from_name ) . '" size="50" />';
 }
 
 function rns_transmissions_from_email_input() {
 	/* Get option 'list_id' value from the database */
 	$options = get_option( 'rns_transmissions_options' );
 	$from_email = $options['from_email'];
 	/* Echo the field */
-	echo "<input type='text' name='rns_transmissions_options[from_email]' id='from_email' value='$from_email' size='50' />";
+	echo '<input type="text" name="rns_transmissions_options[from_email]" id="from_email" value="' . esc_attr( $from_email ) . '" size="50" />';
```
Comment (benlk) [inline includes/admin.php]: This input should probably use `type="email"` instead of `type="text"`

File: includes/admin.php [hunk 4]
```diff
@@ -255,27 +257,31 @@ function rns_transmissions_lists_available_cboxes() {
...
-			$default_group = $options['list_groups'][$ID]['default_group'];
-
-			echo '<br>';
-			echo '<div class="nested-group" style="margin: 10px 0 0 20px;">';
-			foreach ( $available_groups[$ID] as $group ) {
-				echo '<label for="rns_transmissions_options_'. $group['id'] . '_default_group">';
-				echo '<input ' . checked( $group['id'], $default_group, false ) . ' name="rns_transmissions_options[list_groups][' . $ID . '][default_group]" type="radio" id="rns_transmissions_options_' . $group['id'] . '_default_group" value="' . $group['id'] . '" >';
-				echo $group['name'];
-				echo '</label><br>';
+	if ( $available_lists ) {
```
Comment (benlk) [inline includes/admin.php:268]: Does this array ever come back as having items, but only composed of invalid items?

File: includes/admin.php [hunk 5]
```diff
@@ -255,27 +257,31 @@ function rns_transmissions_lists_available_cboxes() {
...
+
+				echo '<br>';
+				echo '<div class="nested-group" style="margin: 10px 0 0 20px;">';
+				foreach ( $available_groups[ $id ]['categories'] as $group ) {
+					echo '<label for="rns_transmissions_options_' . intval( $group['id'] ) . '_default_group">';
+					echo '<input ' . checked( $group['id'], $default_group, false ) . ' name="rns_transmissions_options[list_groups][' . intval( $id ) . '][default_group]" type="radio" id="rns_transmissions_options_' . intval( $group['id'] ) . '_default_group" value="' . intval( $group['id'] ) . '" >';
+					echo esc_html( $group['title'] );
+					echo '</label><br>';
+				}
+				echo '</div>';
```
Comment (benlk) [inline includes/admin.php:286]: This whole section was just an indentation change and not any functionality change, correct?

File: includes/admin.php [hunk 6]
```diff
@@ -102,7 +104,7 @@ function rns_transmissions_create_menu() {
 function rns_transmissions_settings_section_page() {
 	update_available_lists();
 ?>
-  <div class="wrap">
+	<div class="wrap">
```
Comment (benlk) [inline includes/admin.php:116]: Lines between this and `</div>` should be indented to match.

**Review by benlk — COMMENTED** [review]
Minor code style recommendations, and a couple of questions.

File: includes/actions.php (hunk 1 above)
Comment (rclachance) [reply includes/actions.php:154]: use yoda conditions, we must

File: includes/admin.php (hunk 4 above)
Comment (rclachance) [reply includes/admin.php:268]: invalid how?

**Comment by rclachance** [comment]
@benlk made changes based on your feedback, ready for another review

**Pull request merged**